### PR TITLE
feat: add aliases to '--scanners'

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This PR updates the allowed values of `--scanners` from `config` to `misconfig`. It also adds aliases for backward compatibility and spelling discrepancies.

- `misconfig` also allows
  - `config`
  - `misconf`
  - `misconfiguration`
- `vuln` also allows
  - `vulnerability`

### Before

```
$ trivy repo --scanners config .
```

NOTE: It remains working.

### After

```
$ trivy repo --scanners misconfig .
```

## Related Issues
- https://github.com/aquasecurity/trivy/issues/5557

## Related PRs
- https://github.com/aquasecurity/trivy/pull/4990

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/flag/scan_flags.go [hunk 1]
```diff
@@ -37,6 +37,15 @@ var (
 			types.SecretScanner,
 			types.LicenseScanner,
 		}),
+		ValueNormalize: func(s string) string {
```
Comment (DmitriyLewen) [inline pkg/flag/scan_flags.go:40]: Can we add test that this function works as expected?

**Review by DmitriyLewen — COMMENTED** [review]
I left 1 comment.
The rest look good.

But i am worried about different names of target (`config`) and scanner (`misconfig`).
We don't write that we support `config`, `misconf` and `misconfiguration` scanner words and different names can be confusing.
Is there any point in this renaming?

**Comment by knqyf263** [comment]
>But i am worried about different names of target (config) and scanner (misconfig).

'Target' refers to what is being scanned, like container images, VM images, or configuration files. It's the object of our security analysis. 'Scanner', on the other hand, is the tool or method used to detect security issues within the 'Target'. For example, if we scan configuration files, the 'Target' is the configuration files themselves, and the 'Scanner' identifies any misconfigurations in them. Therefore, `trivy config --scanners misconfig` makes sense.

File: pkg/flag/scan_flags.go (hunk 1 above)
Comment (knqyf263) [reply pkg/flag/scan_flags.go:40]: Done
https://github.com/aquasecurity/trivy/pull/5558/commits/6a9343fbbd337d06f6aef5f52af4cf89d4f66e99

**Comment by DmitriyLewen** [comment]
hm... It make sense. You convinced me 😄 
Thanks for explanation.

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Comment by knqyf263** [comment]
`trivy config` is just an alias of `trivy repo/fs --scanners misconfig`. We can remove `trivy config` if it is confusing.

**Comment by DmitriyLewen** [comment]
I think `Trivy config` is convenient option for users who only use Trivy for misconfigurations.

After your explanation, I understand your logic and agree with it.
Also, if user wants to use `scanners` instead of `target` - we support `config` scanner (if he writes `config` by mistake) and this will not be a problem for the user.

I think we don't need to remove `config`.

**Comment by knqyf263** [comment]
OK, I'll wait for @simar7, then.

**Comment by itaysk** [comment]
> We can remove trivy config if it is confusing.

TBH I have always found `trivy config` to be weird command. I think it's fine to have special cases commands which are not targets, but just the name itself "config" is not great. I think it feels more about configuring Trivy rather than scanning something (e.g `git config`). If anything, I'd rename it to `trivy iac` instead of removing. I know this is less "accurate" name, but I believe most users are accustomed to "IaC scanning". not strongly advocating to make a change now, just sharing my thoughts.

**Comment by knqyf263** [comment]
@simar7 This PR is about scanners, not targets. We discussed targets a bit, but this PR renamed `--scanners config` to `--scanners misconfig` and kept `--scanners config` for backward compatibility.

The target (`trivy iac` or `trivy config`) is not a main topic of this PR. We can discuss it later.

I explained the differences between targets and scanners here.
https://github.com/aquasecurity/trivy/pull/5558#issuecomment-1807505053

In short,
- Target (off-topic)
  - config (now)
  - iac
- Scanner (this PR)
  - config (old)
  - misconfig (new)

We used `config` for both the target and scanner, and it was confusing. The word "misconfiguration" is more appropriate for scanners, and the word "IaC" may be more appropriate for targets.

**Comment by simar7** [comment]
> @simar7 This PR is about scanners, not targets. We discussed targets a bit, but this PR renamed `--scanners config` to `--scanners misconfig` and kept `--scanners config` for backward compatibility.
> 
> The target (`trivy iac` or `trivy config`) is not a main topic of this PR. We can discuss it later.
> 
> I explained the differences between targets and scanners here. [#5558 (comment)](https://github.com/aquasecurity/trivy/pull/5558#issuecomment-1807505053)
> 
> In short,
> 
> * Target (off-topic)
>   
>   * config (now)
>   * iac
> * Scanner (this PR)
>   
>   * config (old)
>   * misconfig (new)
> 
> We used `config` for both the target and scanner, and it was confusing. The word "misconfiguration" is more appropriate for scanners, and the word "IaC" may be more appropriate for targets.

Yeah that's fine by me.

**Pull request merged**